### PR TITLE
[REF] Simplify obtuse boolean expressions

### PR DESCRIPTION
**Pull request by colemanw**
Overview
----------------------------------------
Code cleanup - simplifies overly complicated boolean expressions.

I noticed a silly pattern used sometimes in our code where an expression is followed by `? TRUE : FALSE`. This is just an overly verbose way of casting a value to boolean. If the value was already a boolean, it does nothing at all. If the value was something else, the `(bool)` cast is a more concise way of doing the same thing.

I went through these and removed the `? TRUE : FALSE` part of each expression, and added `(bool)` if the value wasn't boolean already.

Before
----------------------------------------
Examples like this...

1. `return ($foo > $bar) ? TRUE : FALSE;`
2. `empty($foo[$bar]) ? TRUE : FALSE;`
3. `isset($foo->$bar) ? TRUE : FALSE;`
4. `return $someScalarVal ? TRUE : FALSE;`

After
----------------------------------------
... were replaced with this:

1. `return $foo > $bar;`
2. `empty($foo[$bar]);`
3. `isset($foo->$bar);`
4. `return (bool) $someScalarVal;`

Technical Details
----------------------------------------
Note that only the 4th example needed the cast added as the others (comparison operators, `isset()` and `empty()`) already return a bool.

### CODE REVIEW

File: CRM/Contact/BAO/Contact/Utils.php [hunk 1]
```diff
@@ -55,11 +55,7 @@ public static function getImage($contactType, $urlOnly = FALSE, $contactId = NUL
         $imageInfo[$contactType]['url'] = $imageUrl;
       }
       else {
-        $isSubtype = (array_key_exists('parent_id', $typeInfo) &&
-          $typeInfo['parent_id']
-        ) ? TRUE : FALSE;
-
-        if ($isSubtype) {
+        if (!empty($typeInfo['parent_id'])) {
```
Comment (colemanw) [inline CRM/Contact/BAO/Contact/Utils.php]: This one takes the prize for overly complicated. 5 lines to check if a value isset, obtusely cast it to a boolean, assign it to a variable (which is only used once) and then read that variable.

File: api/api.php [hunk 2]
```diff
@@ -184,12 +184,7 @@ function _civicrm_api3_api_getfields(&$apiRequest) {
  *   true if error, false otherwise
  */
 function civicrm_error($result) {
-  if (is_array($result)) {
-    return (array_key_exists('is_error', $result) &&
-      $result['is_error']
-    ) ? TRUE : FALSE;
-  }
-  return FALSE;
+  return is_array($result) && !empty($result['is_error']);
```
Comment (colemanw) [inline api/api.php]: Actually this one comes in first-place :trophy:. The other will have to take runner-up.

**Comment by eileenmcnaughton** [comment]
I reviewed some of these & submitted them & merged them

**Comment by totten** [comment]
`r-code`: I read a bunch of these, and they looked equivalent. :+1:

IMHO, it is good breaking these up into smaller chunks (as in @eileenmcnaughton's extracted PRs). A few reasons:

* It's hard to be perfect -- with 120-130 changes,  and with the subtle variations of true-ish-ness, the odds are that some kind of mistake slips through our notice.
* Our error-rate in reading is probably better with more mid-sized batches.
* If there is a problem, we'll probably need to do some kind of bisecting - which is also easier with smaller batches.
* If there is a problem, someone may try reverting a commit (eg for diagnostic purposes). But if the commit is large, then the odds grow that some intervening change makes it conflicted.

For this type of change, committing each folder usually gives mid-sized chunks, e.g

```bash
for DIR in CRM/Core CRM/Utils CRM/Member ; do git commit "$DIR" -m "$DIR - Cleanup booleans" ; done
```

**Comment by colemanw** [comment]
test this please

**Comment by colemanw** [comment]
I've broken this out into smaller PRs per @eileenmcnaughton's request.

**Pull request closed without merging**